.github/workflows/project_tests.yml

```diff
@@ -0,0 +1,63 @@
+name: Project tests
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        engine:
+          - libfuzzer
+        sanitizer:
+          - address
+          - memory
+          - undefined
+        architecture:
+          - x86_64
+        include:
+          - engine: afl
+            sanitizer: address
+            architecture: x86_64
+          - engine: honggfuzz
+            sanitizer: address
+            architecture: x86_64
+          - engine: libfuzzer
+            sanitizer: address
+            architecture: i386
+          - engine: none
+            sanitizer: address
+            architecture: x86_64
+          - engine: dataflow
+            sanitizer: dataflow
+            architecture: x86_64
+    env:
+      ENGINE: ${{ matrix.engine }}
+      SANITIZER: ${{ matrix.sanitizer }}
+      ARCHITECTURE: ${{ matrix.architecture }}
+
+    steps:
+    - uses: actions/checkout@v2
+    - run: |  # Needed for git diff to work.
+        git fetch origin master --depth 1
+        git symbolic-ref refs/remotes/origin/HEAD refs/remotes/origin/master
+
+    - name: Clear unnecessary files
+      run: |
+        sudo swapoff -a
```

Comment:
these seem pretty hacky. Are they really needed? e.g. do we need "apt clean" if we don't install any apt packages?

Reply:
This is directly coming as a recommendation from GitHub Actions support channel (we use this in FuzzBench as well). See https://github.community/t/bug-strange-no-space-left-on-device-ioexceptions-on-github-runners/17616/9